collector/textfile.go

```diff
@@ -249,3 +242,19 @@ fileLoop:
 	)
 	return nil
 }
+
+// helper: returns true if file contains unsupported timestamps
+func checkUnsupportedTimestams(parsedFamilies map[string]*dto.MetricFamily, path string) bool {
```

Comment:
The function name `checkUnsupportedTimestams()` is misleading (besides missing a "p"), as the function does more than checking timestamps (it auto-fills missing help strings as well).

Maybe still keep the existing loop in `Collect()` just for the help-autofilling, but pull out the timestamp-checking into a separate function? It does mean looping over all parsed metric families twice, but I think that's ok.

Reply:
@juliusv ok i will have look thx